packages/tools/devtools/devtools-core/src/data-visualization/DataEditing.ts

```diff
@@ -0,0 +1,43 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ISharedObject } from "@fluidframework/shared-object-base";
+import { Serializable } from "@fluidframework/datastore-definitions";
+import { EditType } from "../CommonInterfaces";
+
+// Ideas:
+// - Hold onto previous summary and only transmit diff?
+
+// TODOs:
+// - Dependency tracking
+//   - When a particular DDS is no longer reachable via the input data, we need to remove it from the map and stop
+//     emitting updates.
+
+/**
+ * The type of a shared object.
+ *
+ * @remarks
+ *
+ * This can be acquired via {@link @fluidframework/datastore-definitions#IChannelFactory.Type} field of
+ * your shared object's factory class.
+ */
+export type SharedObjectType = string;
+
+/**
+ * Generates a visual description of the provided {@link @fluidframework/shared-object-base#ISharedObject}'s
```

Comment:
Let's update the comments to reflect editing 😜

Reply:
👍